### ISSUE TITLE
linter: extension stops reporting errors in vscode

### ISSUE THREAD

**Issue by yzhe554** [oxc-project/oxc#19244]
### What version of Oxlint are you using?

oxlint 1.46.0 and extension 1.43.0

### What does your `.oxlintrc.json` config file look like?

```jsonc
// your config here
"typescript/prefer-optional-chain": "error",
```

### What happened?

In oxlint 1.43.0 and extension 1.43.0 error is available in vs code

<img width="605" height="59" alt="Image" src="https://github.com/user-attachments/assets/f143b1ff-63d1-47ac-9a7f-27f65bba1cc5" />

It stops working after oxlint is upgraded to 1.46.0, regardless of the extension version(1.43.0 or 1.45.0)

**Comment by connorshea** [comment]
Can you share the code directly? (or a censored version of the code that reproduces the issue, if it's proprietary). We intentionally changed the behavior of prefer-optional-chain in https://github.com/oxc-project/tsgolint/pull/606, which may be related.

If this does not trigger a violation in the original ESLint rule, then the previous behavior was a bug, and this should not render an error.

Does the violation (or any violations for any rules) appear via running oxlint in your terminal? Are you certain this is a VS Code-specific issue?

**Comment by connorshea** [comment]
Also, how are you enabling type-aware rules? The `typeAware` flag in VS Code, or something else?

**Comment by yzhe554** [comment]
@connorshea I have enabled all the config. Otherwise, how could it work in old version?

It stops working for me for any rules.

`const age: any = 'seventeen';`

<img width="719" height="789" alt="Image" src="https://github.com/user-attachments/assets/4f1ab9ae-d717-4fdf-a209-99f6c4d50732" />

**Comment by yzhe554** [comment]
@connorshea I take a deep look into the logs. It seems to be js-plugins related.
If I disable all the js-plugins I have created, it works again.
TypeError [ERR_UNKNOWN_FILE_EXTENSION]: Unknown file extension \".ts\"

I have switched to @oxlint/plugins, not sure why there is still an error.

Also why it works in cli but not in extension?

**Comment by connorshea** [comment]
Hmm, do you have `@oxlint/plugins` installed then? And what version of `oxlint-tsgolint`?

If you're able to provide the output from the VS Code output channel that'd be appreciated as well, I tried reproducing the issue with no-explicit-any failing in VS Code, but it works fine for me on latest oxlint+VS Code extension. Does the custom JS Plugin you've written have any differences vs. the docs describe here? https://github.com/oxc-project/oxc/blob/31b562f70baf3562264e34df16cbad2e096edc40/npm/oxlint-plugins/README.md

**Comment by connorshea** [comment]
Also, are you able to provide the full oxlint config file? Is this Windows, macOS, or Linux?

**Comment by yzhe554** [comment]
@connorshea  please use this reproducible repo https://github.com/yzhe554/oxlint-nextjs/blob/main/test.ts

It's for any rules rather than just type aware ones

**Comment by Yizack** [comment]
VSCode extension fails to start when `jsPlugins` is present in `.oxlintrc.json`. CLI works correctly. No error appears in the output panel. Removing `jsPlugins` works.

<details>
  <summary>
    <code>index.js</code>
  </summary>

```js
console.log("Hello from Oxlint");
```
</details>

<details>
  <summary>
    <code>.oxlintrc.json</code>
  </summary>

```json
{
  "ignorePatterns": ["node_modules/"],
  "jsPlugins": [
    "@stylistic/eslint-plugin"
  ],
  "rules": {
    "no-console": "error",
    "@stylistic/semi": ["error", "never"]
  }
}
```
</details>


<details>
  <summary>
    <code>package.json</code>
  </summary>

```json
{
  "name": "oxlint-test",
  "type": "module",
  "private": true,
  "scripts": {
    "lint": "oxlint"
  },
  "devDependencies": {
    "@stylistic/eslint-plugin": "^5.8.0",
    "oxlint": "^1.46.0"
  }
}
```
</details>

VsCode about:
```
Version: 1.109.2 (system setup)
Commit: 591199df409fbf59b4b52d5ad4ee0470152a9b31
Date: 2026-02-10T20:18:23.520Z
Electron: 39.3.0
ElectronBuildId: 13168319
Chromium: 142.0.7444.265
Node.js: 22.21.1
V8: 14.2.231.22-electron.0
OS: Windows_NT x64 10.0.26200
```

Edit:

I just noticed that if I:

1. Open VSCode with a config without `jsPlugins` -> extension works
2. Add `jsPlugins` and js plugin rules
3. Save the config file -> Then the extension and js plugin rules start working correctly after saving (only works with `oxlintrc.json`, not with `oxlint.config.ts`
5. But if I click on Restart server -> extension stops working completely

**Comment by connorshea** [comment]
> [@connorshea](https://github.com/connorshea) please use this reproducible repo https://github.com/yzhe554/oxlint-nextjs/blob/main/test.ts
> 
> It's for any rules rather than just type aware ones

Well, the bad news is that it works fine for me on macOS:

<img width="508" height="254" alt="Image" src="https://github.com/user-attachments/assets/37b7d4c0-a7ea-4ea0-9474-9e70eb2a9522" />

Even the custom plugin:

<img width="948" height="336" alt="Image" src="https://github.com/user-attachments/assets/ab5074d8-407f-4666-bc34-9e4ca3eb3a5a" />

The good news is that this means we have narrowed it down to what is almost certainly a windows-specific bug somewhere. I just... don't know where.

**Comment by connorshea** [comment]
@Sysix @camc314 here's a new one, for your enjoyment

**Comment by yzhe554** [comment]
> > [@connorshea](https://github.com/connorshea) please use this reproducible repo https://github.com/yzhe554/oxlint-nextjs/blob/main/test.ts
> > It's for any rules rather than just type aware ones
> 
> Well, the bad news is that it works fine for me on macOS:
> 
> <img alt="Image" width="508" height="254" src="https://private-user-images.githubusercontent.com/2977353/547999631-37b7d4c0-a7ea-4ea0-9474-9e70eb2a9522.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzA3Njk5OTEsIm5iZiI6MTc3MDc2OTY5MSwicGF0aCI6Ii8yOTc3MzUzLzU0Nzk5OTYzMS0zN2I3ZDRjMC1hN2VhLTRlYTAtOTQ3NC05ZTcwZWIyYTk1MjIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI2MDIxMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNjAyMTFUMDAyODExWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MjY4OWVmODFmNDNjNDYxNjU5ZWMwZTE4ZDQ4MmY3MTI2OWRmMjk4ZjUyNDdjNGRlZWM1NmNiYmMyYThkZDg2MCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.45aAhRiE3aBh5ZzumRUyRTwE_j2sJ-gk0ZHjYRTG5RY">
> Even the custom plugin:
> 
> <img alt="Image" width="948" height="336" src="https://private-user-images.githubusercontent.com/2977353/547999877-ab5074d8-407f-4666-bc34-9e4ca3eb3a5a.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzA3Njk5OTEsIm5iZiI6MTc3MDc2OTY5MSwicGF0aCI6Ii8yOTc3MzUzLzU0Nzk5OTg3Ny1hYjUwNzRkOC00MDdmLTQ2NjYtYmMzNC05ZTRjYTNlYjNhNWEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI2MDIxMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNjAyMTFUMDAyODExWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ODMyMTNmYzUzY2MyMTEwYjY5NjZiNzllMWUxMzc3M2E2OGJkMWEwMGEyOWFkYTM1NjY4ZDU0N2VkNzlkMjBlYiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.IxXIKh7ywvXPQGtReF3-afdHyso0Kn9vXyqGOd5DBKg">
> The good news is that this means we have narrowed it down to what is almost certainly a windows-specific bug somewhere. I just... don't know where.

Hmmm, I am on mac as well. with node 24.13.0...

**Comment by yzhe554** [comment]
Share the full error logs

<img width="837" height="341" alt="Image" src="https://github.com/user-attachments/assets/6d82fd0d-8604-4f0a-a0e1-24414baf7dd9" />

**Comment by yzhe554** [comment]
The issue is likely to be the extension cannot load any ts files anymore
`Unknown file extension \".ts\"`

**Comment by yzhe554** [comment]
For whoever has the same issue, the work around is to build the ts into js.
The following is a working config for rolldown
```
import { defineConfig } from 'rolldown';
/**
 * this is a temporary solution when oxc extension is broken with ts files
 */

export default defineConfig({
	input: 'js-plugins/index.ts',
	output: {
		file: 'js-plugins/index.js',
		format: 'esm',
	},
	resolve: {
		extensions: ['.ts', '.js'],
		mainFields: ['module', 'main'],
		conditionNames: ['import', 'require', 'node', 'default'],
	},
	platform: 'node',
	treeshake: true,
	external: ['@oxlint/plugins'],
});

```

**Comment by camc314** [comment]
> TypeError [ERR_UNKNOWN_FILE_EXTENSION]: Unknown file extension ".ts"

This error means that oxlint is running with a node version that doesn't support importing TypeScript files.

I suspect that vscode is using some sort of different node version vs the CLI which is causing this.

If you open up activity monitor, can you inspect and see what node versions are running?

**Comment by camc314** [comment]
@Sysix can we add a log inside either the vscode extension or the language server to improve debugging this that just logs the node version that we're using?

**Comment by yzhe554** [comment]
> > TypeError [ERR_UNKNOWN_FILE_EXTENSION]: Unknown file extension ".ts"
> 
> This error means that oxlint is running with a node version that doesn't support importing TypeScript files.
> 
> I suspect that vscode is using some sort of different node version vs the CLI which is causing this.
> 
> If you open up activity monitor, can you inspect and see what node versions are running?

I don't get why this isn't an issue in OXLint/extension 1.43.0. It seems a breaking change from the new version. I'm pretty sure I use node 24.13.0 as we'll manage the version via nvm/mise

**Comment by camc314** [comment]
We added support for JS plugins in the language server in this release, and that's the part that's erroring.

Let me see how we handle failures to load these plugins

**Comment by yzhe554** [comment]
> We added support for JS plugins in the language server in this release, and that's the part that's erroring.
> 
> Let me see how we handle failures to load these plugins

My 2 cents, instead of using same node versions, would it be possible that extension automatically compiles ts to js. Sorry I'm not too familiar with extensions.

**Comment by yzhe554** [comment]
Tested the issue still exists after https://github.com/oxc-project/oxc/issues/19265

**Comment by overlookmotel** [comment]
@yzhe554 I believe this is a separate issue from #19265. This one relates to plugins which are `.ts` files. Oxlint 1.47.0 should have fixed the more general problem with JS plugins.

Can I just check that with your workaround (converting plugins to `.js` files) still solves the issue on 1.47.0?

That's not to say that the lack of support for `.ts` files isn't a problem, just making sure we understand the scope of the issue.

**Comment by yzhe554** [comment]
> @yzhe554 I believe this is a separate issue from #19265. This one relates to plugins which are `.ts` files. Oxlint 1.47.0 should have fixed the more general problem with JS plugins.
> 
> Can I just check that with your workaround (converting plugins to `.js` files) still solves the issue on 1.47.0?
> 
> That's not to say that the lack of support for `.ts` files isn't a problem, just making sure we understand the scope of the issue.

Yeh, I can confirm js workaround still works. I just mentioned as I see it's referenced. Sorry if that confused you.

**Comment by yzhe554** [comment]
Try to follow up with this one

**Comment by csvn** [comment]
I'm not sure if this issue is the same as mine, but it seems similar.

I'm on Windows and not seeing errors in the editor via the VSCode extension/LSP. Can sometimes trigger errors to show via command `oxc: Fix all auto-fixable problems (file)`. Currently using two JS plugins:

```jsonc
// .oxlintrc.json
{
  "jsPlugins": ["@angular-eslint/eslint-plugin", "@stylistic/eslint-plugin"],
  // ...
}
```

I do see some errors in my config file though:

```
2026-04-01 18:15:52.956 [info] 2026-04-01T16:15:52.955991Z  WARN oxlint::lsp::server_linter: Skipping config
file c:/dev/project\.oxlintrc.jsonc: Parse { path: "c:/dev/project\\.oxlintrc.jsonc", error: OxcDiagnostic
{ inner: OxcDiagnosticInner { message: "Failed to parse jsonc file c:/dev/project\\.oxlintrc.jsonc: Kind(InvalidData)",
labels: None, help: None, note: None, severity: Error, code: OxcCode { scope: None, number: None }, url: None } } }

2026-04-01 18:19:05.648 [info] 2026-04-01T16:19:05.647666Z  WARN oxlint::lsp::server_linter: Failed to
build config from oxlintrc: Failed to load JS plugin: @angular/eslint-plugin
  Cannot find module '@angular/eslint-plugin'
```

I've no issues running `npx oxlint` though, so seems to only be via the extension/LSP.

I can create a separate issue if this is not the same root problem.
